libsolidity/ast/Types.cpp

```diff
@@ -2919,13 +2919,15 @@ FunctionType::FunctionType(ErrorDefinition const& _error):
 		m_parameterTypes.push_back(var->annotation().type);
 	}
 
+	m_returnParameterNames.push_back("");
+	m_returnParameterTypes.push_back(TypeProvider::magic(MagicType::Kind::Error));
+
 	solAssert(
 		m_parameterNames.size() == m_parameterTypes.size(),
 		"Parameter names list must match parameter types list!"
 	);
 	solAssert(
-		m_returnParameterNames.size() == 0 &&
-		m_returnParameterTypes.size() == 0,
+		m_returnParameterNames.size() == m_returnParameterTypes.size(),
 		""
 	);
 }
```

Comment:
You can also remove it - it was downright paranoid already... the other one as well...